community/kernel/src/main/java/org/neo4j/kernel/impl/coreapi/TxStateTransactionDataSnapshot.java

```diff
@@ -219,6 +222,10 @@ private void takeSnapshot()
                                         properties.get().value() ) );
                             }
                         }
+                        catch ( PropertyKeyIdNotFoundKernelException e )
+                        {
+                            throw new IllegalStateException( "An entity that does not exist was modified.", e );
```

Comment:
we also need to include entity id to message since we will not gonna see that from stacktrace